lib/puppet/graph/simple_graph.rb

```diff
@@ -146,7 +146,7 @@ def tarjan(root, s)
         frame[:step] = :children
 
       else
-        fail "#{frame[:step]} is an unknown step"
+        fail _("%{step} is an unknown step") % { step: frame[:step] }
```

Comment:
I'm not sure we should externalize this. It's such an obscure error in the graph logic that someone who comes across it in japanese will probably have better luck searching for resolution in english than with a translation that probably won't reflect the nature of the message.